### DISCUSSION TITLE
navigator.hid not available, blocking ledger support.

### DISCUSSION THREAD

**Discussion by barthuijgen** [MetaMask/snaps#2632] [General]
It currently seems impossible to implement ledger support in a snap because `navigator` and therefore `navigator.hid` is not available.

There is an example repository: [https://github.com/MetaMask/ledger-hardware-wallet-snap](https://github.com/MetaMask/ledger-hardware-wallet-snap)
But this does not work. It seems to be about a year old, not sure if anything changed in the meantime that broke it.

The repository uses `"endowment:webhid": {}` but this permission does not exist.

Reproduction steps: try using `navigator` in any snap's `onRpcRequest` method, and it should be undefined. At least it is for me.

**Comment by roo-shy** [comment]
Hi @barthuijgen snaps don't support WebHID at the moment

**Comment by barthuijgen** [comment]
Hi @roo-shy Thanks for the answer, just some follow up questions.

1. Why is there an example repo showing how to use ledger but it is not supported?
2. How do you suggest developers integrate ledger into a snap wallet currently?

**Reply by Mrtenz** [reply]
Hi @barthuijgen, the hardware wallet Snap you linked was a proof of concept/experimentation at the time. This functionality was never actually implemented by the Snaps platform. We should clarify this in the README of the repo as well.

Right now there is no way for Snaps to use hardware wallets, as supporting this in all the browsers supported by MetaMask is technically very challenging. We are looking into ways to solve this however, but we don't have any timeline on this at the moment.
